features/poll.c

```diff
@@ -0,0 +1,46 @@
+// This detects whether a) poll() is available, and b) it works as expected
```

Comment:
Yes. The legacy branch didn't need it because it simply didn't use `poll()`; at least not on macOS. And, unsurprisingly, the legacy IFFE test for whether `poll()` is available returns true. There is something else in the headers that causes it to be #undef'ed.